### PR TITLE
initial docker functionality

### PR DESCRIPTION
**Pull request by slavaaaaaaaaaa**
untested initial commit. waiting for docker to compile on gentoo

woooo

kill me

### CODE REVIEW

File: bin/setup-centos7.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ rkhunter --propupd
 
 if [ ! -d "/opt/argon2" ]; then
 	mkdir -p /opt/argon2
-        git clone https://github.com/P-H-C/phc-winner-argon2.git /opt/argon2
+        git clone https://github.com/P-H-C/phc-winner-argon2.git /opt/argon2 || exit 1
```
Comment (jaypeah2) [inline bin/setup-centos7.sh]: what is this for?

Comment (slavaaaaaaaaaa) [reply bin/setup-centos7.sh]: if we can't clone the repo (e.g. github is down), we shouldn't continue

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM centos:7
+
+EXPOSE 10000 10080 10443
+
+COPY . /libjaypea
+
+RUN ["/libjaypea/bin/setup-centos7.sh"]
+
+RUN groupadd -r -g 1000 user && useradd -r -m -g user -u 1000 user
```
Comment (jaypeah2) [inline Dockerfile:22]: could you tell me about these options? ill look into them

Comment (slavaaaaaaaaaa) [reply Dockerfile:22]: https://docs.docker.com/engine/reference/builder/ basically. Run will run the specified command, entrypoint will do some other janky stuff

**Comment by slavaaaaaaaaaa** [comment]
I just realized something; we're going to have to refactor. 

There's no reason to have systemd within docker.  We simply need to write a systemd docker container unit and run the container with it. Bam. 

We also don't need to have firewalld and fail2ban in the same container. Proper usage would be to split those into two separate containers, ending up with three containers. At that point, docker-compose should be implemented.

**Comment by slavaaaaaaaaaa** [comment]
whose fucking idea was this
fuck docker

**Pull request closed without merging**